### ISSUE TITLE
user homes on btrfs should always be a subvolume

### ISSUE THREAD

**Issue by cmurf** [systemd/systemd#15121]
**systemd version the issue has been seen with**
> systemd-245-1.fc33.x86_64

**Used distribution**
 > Fedora

**Expected behaviour you didn't see**
 > Btrfs on LUKS, the user home is a subvolume.

**Unexpected behaviour you saw**
> Btrfs on LUKS, the user home is a directory.

**Steps to reproduce the problem**
```
# homectl create --disk-size=5G --fs-type=btrfs chris`
# homectl activate chris`
# mount
/dev/mapper-chris on /home/chris type btrfs (rw,nosuid,nodev,subvolid=5,subvol=/chris)
```
There is no subvolume chris on this file system; chris is a directory and is bind mounted to /home/chris; and therefore /home/chris cannot be snapshot.

Whereas
```
# homectl create --storage=subvolume chris`
# homectl activate chris`
# mount
/dev/mmcblk0p3 on /home/chris type btrfs (rw,nosuid,nodev,noatime,compress=zstd,nossd,space_cache=v2,subvolid=298,subvol=/home/chris.homedir)
```
Here, chris.homedir is a subvolume, and therefore /home/chris can be snapshot.

--storage=
Takes one of "luks", "fscrypt", "directory", "subvolume", "cifs".

When --storage=luks and --fs-type=btrfs, this should also imply storage subvolume so that the user home is a subvolume rather than a directory. That way it can be snapshot. 

This could be related to #15120. Additionally it would become necessary to mount the subvolume direction upon activation, i.e. `mount -o subvol=chris /dev/mapper/chris /home/chris` just like when --storage=subvolume

Snapshotting the user home subvolume is 
a) faster than reflink copying the loop file
b) always available, where the backing file system might be ext4 which doesn't support reflink
c) chattr +C nocow files on Btrfs become COW if they are reflink copied, suboptimal
d) reflink copies will complicate free space reporting
e) logically consistent command regardless of whether storage type is subvolume or Btrfs on LUKS; simply:
`btrfs sub snap -r /home/chris /home/chris/.snapshots/chris.20200315T0024Z`

**Comment by poettering** [comment]
Yeah, I guess we should use a subvol inside btrfs LUKS homedirs. true.

**Comment by poettering** [comment]
Fix waiting in #15376
